chain/solana/solana.go

```diff
@@ -56,17 +97,25 @@ func (client *Client) CallContract(ctx context.Context, contract address.Address
 		return pack.Bytes(nil), fmt.Errorf("decoding result: empty")
 	}
 
-	// Decode the data associated with the account into pack-encoded bytes.
+	// Deserialise the account's info into the appropriate struct.
 	info := ResponseGetAccountInfo{}
 	if err := json.Unmarshal(*res.Result, &info); err != nil {
 		return pack.Bytes(nil), fmt.Errorf("decoding result: %v", err)
 	}
-	fmt.Printf("account data: %v", info.Value.Data)
 
+	// Decode the Base58 encoded account data into raw byte-representation. Since
+	// this holds the burn log's data, it should be 41 bytes in size.
+	//
+	// burn amount: 8 bytes
+	// recipient length: 1 byte
+	// recipient: 32 bytes
 	data := base58.Decode(info.Value.Data)
-	// data, err := base64.StdEncoding.DecodeString()
 	if err != nil {
 		return pack.Bytes(nil), fmt.Errorf("decoding result from base58: %v", err)
 	}
+	if len(data) != 41 {
+		return pack.Bytes(nil), fmt.Errorf("unexpected data length. Expected 41, got: %v", len(data))
+	}
```

Comment:
This still feels application-specific. Other non-RenVM applications probably do not have this restriction.